### PR TITLE
tenant_id checking reuse current controller logic

### PR DESCRIPTION
**Pull request by akostadinov**
It turned out that we didn't cover HTTP basic auth use case previously
in THREESCALE-10686.

But also figured out we can reuse the logic inside current controller
instance instead of reimplementing it inside the checker.

### CODE REVIEW

File: test/integration/multitenant_enforcement_test.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ class MultitenantEnforcementTest < ActionDispatch::IntegrationTest
     plan.update_column(:tenant_id, @provider.tenant_id + 1)
     get admin_service_application_plans_path(service)
   rescue => exception
+    assert_predicate exception, :cause
```
Comment (jlledom) [inline test/integration/multitenant_enforcement_test.rb:17]: Why is this needed?

File: test/integration/multitenant_enforcement_test.rb [hunk 2]
```diff
@@ -54,13 +55,33 @@ class MultitenantEnforcementTest < ActionDispatch::IntegrationTest
...
+    auth_pair << [token, ""]
+    auth_pair << ["", master_account.provider_key]
+    auth_pair << [master_account.provider_key, ""]
+
+    auth_pair.each do |pair|
+      basic_auth_str = ActionController::HttpAuthentication::Basic.encode_credentials(*pair)
+      get admin_api_service_application_plans_path(service_id: service.id, format: :json), headers: { 'HTTP_AUTHORIZATION' => basic_auth_str }
+      assert_response :success
+    end
+  end
```
Comment (jlledom) [inline test/integration/multitenant_enforcement_test.rb:83]: I think I need an explanation for this scenario. So there is a plan belonging to a service but with different `tenant_id`. That looks wrong, why must it be allowed when the request is authenticated with HTTP basic auth?

File: test/integration/multitenant_enforcement_test.rb (hunk 1 above)
Comment (akostadinov) [reply test/integration/multitenant_enforcement_test.rb:17]: if `cause` is `nil`, then we don't have any indication in error output what was `exception`. With this assertion, we will see at least what was `exception`.

File: lib/three_scale/middleware/multitenant.rb [hunk 3]
```diff
@@ -57,44 +57,22 @@ def verify!(object)
...
-          param_places = %w[action_dispatch.request.query_parameters action_dispatch.request.request_parameters]
-          possible_keys = %w[provider_key api_key]
-          param_places.product(possible_keys).any? { |params, key| @env[params][key] == @master.provider_key }
-        end
-
-        def token_of_master?
-          token = @env["action_dispatch.request.query_parameters"]["access_token"] ||
-            @env["action_dispatch.request.request_parameters"]["access_token"]
-          @master.access_tokens.find_from_value(token)
+          @is_master ||= @env["action_controller.instance"].send(:current_account)&.master?
```
[the remark this answers is not in the document]
Comment (akostadinov) [reply lib/three_scale/middleware/multitenant.rb:75]: I assume that the controller instance is one of the first things to be initialized on a request. While in `TenantChecker` we can reach this part of the code only after at least 2 ActiveRecord objects are initialized. And any operations that would cause active record objects initialization (creating new objects or retrieving from db) must happen within the controller code. So I think it should always be initialized at this point.

File: test/integration/multitenant_enforcement_test.rb (hunk 2 above)
Comment (akostadinov) [reply test/integration/multitenant_enforcement_test.rb:83]: > **multitenant master** can retrieve from multiple tenants by http basic auth

There are other scenarios that are the same except "by provider key", "by token", "by api_key".

Master must be allowed to see anything.

**Review by jlledom — APPROVED** [review]
